pkg/3scale/amp/operator/configmap_reconciler.go

```diff
@@ -3,77 +3,76 @@ package operator
 import (
 	"context"
 	"fmt"
-	"reflect"
 
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 )
 
-type ConfigMapReconciler struct {
-	BaseReconciler
-	ObjectMetaMerger ObjectMetaMerger
+type ConfigMapReconciler interface {
+	IsUpdateNeeded(desired, existing *v1.ConfigMap) bool
 }
 
-func NewConfigMapReconciler(baseReconciler BaseReconciler, objectMetaMerger ObjectMetaMerger) ConfigMapReconciler {
-	return ConfigMapReconciler{
-		BaseReconciler:   baseReconciler,
-		ObjectMetaMerger: objectMetaMerger,
+type ConfigMapBaseReconciler struct {
+	BaseAPIManagerLogicReconciler
+	reconciler ConfigMapReconciler
+}
+
+func NewConfigMapBaseReconciler(baseAPIManagerLogicReconciler BaseAPIManagerLogicReconciler, reconciler ConfigMapReconciler) *ConfigMapBaseReconciler {
+	return &ConfigMapBaseReconciler{
+		BaseAPIManagerLogicReconciler: baseAPIManagerLogicReconciler,
+		reconciler:                    reconciler,
 	}
 }
 
-func (r *ConfigMapReconciler) Reconcile(desiredConfigMap *v1.ConfigMap) error {
-	objectInfo := ObjectInfo(desiredConfigMap)
-	existingConfigMap := &v1.ConfigMap{}
-	err := r.Client().Get(context.TODO(), types.NamespacedName{Name: desiredConfigMap.Name, Namespace: desiredConfigMap.Namespace}, existingConfigMap)
+func (r *ConfigMapBaseReconciler) Reconcile(desiredCM *v1.ConfigMap) error {
```

Comment:
I understand the followed criteria to reconcile ConfigMaps/Secrets has been to reconcile the fields of the ones that are indirectly set through a CR field.
Not for this PR, but we should talk about what are we going to do with the reconciliation of ConfigMaps/Secrets.